shared/bash_remediation_functions/variables.sh

```diff
@@ -0,0 +1,28 @@
+# Functions for dealing with variables and common operations on them.
+
+function normalize_boolean() {
+  local value="${1,,}"
+  local default="${2:-false}"
+
+  if [ "$value" == "true" ] || [ "$value" == "yes" ]; then
+    echo "true"
+  elif [ "$value" == "false" ] || [ "$value" == "no" ]; then
+    echo "false"
+  else
+    echo "$default"
+  fi
+}
+
+function normalize_state() {
```

Comment:
I think that as we are in 100% control of usage of the function, why not to enforce its arguments instead of normalizing them?

Reply:
@cipherboy What are you worried with that requires these normalizing functions?

Reply:
Sorry folks. I'm on PTO and likely won't have time to address these PR comments until Friday (I go back to work officially on Monday). :/ 

I was trying to make this more "Ansible"-like to make writing Bash remediations more friendly to non-programmers (or even those not familiar with bash). 

IIRC, Ansible handles several variations of the Booleans (`yes`/`no`/`true`/`false`/`True`/`False`/...). These differences aren't caught at compile time (in Bash remediations) because we don't execute them. It would be up to PR author to discover how our module works and ensure they use it correctly, and up to the PR reviewers that they double check all arguments are correct &c.. ...and potentially run the test suite, when present for a rule, to ensure the remediation works as expected.

Seems like we could be "friendly" (for new contributors) and accept more options rather than fewer. Especially for high-level helper functions that we're hoping for new contributors to use. My 2c.

Reply:
I now understand the motivation, and I am still in favor of simple strict interface vs "friendly" interface. To some extent, this is a matter of taste, especially normalizing those booleans. However, present/absent are pretty clear.